File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] n) {
        int res[]=new int[n.length];
        int a=1;
        res[0]=1;
        for(int i=1;i<n.length;i++){
            res[i]=res[i-1]*n[i-1];
        }
        for(int i=n.length-1;i>=0;i--){
            res[i]=res[i]*a;
            a=a*n[i];
        }
        return res;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, n: List[int]) -> List[int]:
        z=0
        p=1
        for i in n:
            if i==0:
                z+=1
            else:
                p*=i
        if(z>1):
            return [0]*len(n)
        if(z==1):
            for i in range(len(n)):
                if n[i]==0:
                    n[i]=p
                else:
                    n[i]=0
        else:
            for i in range(len(n)):
                n[i]=int(p/n[i])
        return n